modules/light-clients/07-tendermint/types/update_test.go

```diff
@@ -367,6 +167,228 @@ func (suite *TendermintTestSuite) TestCheckHeaderAndUpdateState() {
 	}
 }
 
+func (suite *TendermintTestSuite) TestVerifyHeader() {
+	var (
+		path   *ibctesting.Path
+		header *ibctmtypes.Header
+	)
+
+	// Setup different validators and signers for testing different types of updates
+	altPrivVal := ibctestingmock.NewPV()
+	altPubKey, err := altPrivVal.GetPubKey()
+	suite.Require().NoError(err)
+
+	revisionHeight := int64(height.RevisionHeight)
+
+	// create modified heights to use for test-cases
+	altVal := tmtypes.NewValidator(altPubKey, revisionHeight)
+	// Create alternative validator set with only altVal, invalid update (too much change in valSet)
+	altValSet := tmtypes.NewValidatorSet([]*tmtypes.Validator{altVal})
+	altSigners := getAltSigners(altVal, altPrivVal)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			name:     "success",
+			malleate: func() {},
+			expPass:  true,
+		},
+		{
+			name: "successful verify header for header with a previous height",
```

Comment:
maybe add a couple comments on what this test is doing. Will be hard to understand when you come back a year later :)